addons/hr_holidays/hr_holidays.py

```diff
@@ -325,6 +325,8 @@ def create(self, cr, uid, values, context=None):
         """ Override to avoid automatic logging of creation """
         if context is None:
             context = {}
+        partner_id = self.pool['hr.employee'].browse(cr, uid, values['employee_id']).user_id.partner_id
+        values['message_follower_ids'] = [(4, partner_id.id)]
```

Comment:
there is helper methods in `mail_thread.py` like `message_subscribe` you can use